pages/playlists/[id].tsx

```diff
@@ -1,39 +1,69 @@
-import PlaylistTracks from '@/components/PlaylistTracks';
+import TrackCard from '@/components/TrackCard';
 import PlaylistDetails from '@/components/PlaylistDetails';
 import DashboardLayout from '@/layouts/DashboardLayout';
 import { APIRoute } from '@/models/APIRoute.enum';
-import { SpotifyPlaylist } from '@/models/Spotify';
+import { SpotifyPlaylist, PlaylistTrack } from '@/models/Spotify';
 import { NextPage } from 'next';
 import { useRouter } from 'next/router';
 import SkeletonList from '@/components/SkeletonList';
 import useSWR from 'swr';
+import usePaginatedData from '@/lib/usePaginatedData';
+import Button from '@/components/Button';
 
 const Playlist: NextPage = () => {
   const { query } = useRouter();
+  const {
+    data,
+    size,
+    setSize,
+    isReachingEnd,
+    isLoadingInitialData,
+    isLoadingMore,
+  } = usePaginatedData<PlaylistTrack>({
+    url: `${APIRoute.PLAYLISTS}/${query.id}/tracks` || null,
```

Comment:
This is not correct, this will always return `true` since it's a `string`. Needs to be

```suggestion
    url: query.id ? `${APIRoute.PLAYLISTS}/${query.id}/tracks` : nulll,
```